File: XMLCheck/Handler/CheckRulesXML03/Xml3RuleBase.cs
using XMLCheck.Models;
using XMLCheck.Services;

namespace XMLCheck.Handler.CheckRulesXML03
{
    public abstract class Xml3RuleBase:IXml3Rule
    {
        public abstract string Key { get; }

        public abstract ErrorDetails? Check(Xml3Model model,HoSoContext context);

        protected ErrorDetails Error(string message)
        {
            return new ErrorDetails
            {
                key = Key,
                message = message
            };
        }
    }
}

File: XMLValidator_v2/src/parsers/rules/xml3/xml3RuleBase.ts
import { Xml3Model } from '../../../interfaces/xml-models.interface';
import { HoSoContext } from '../../../services/hoso-context';
import { ErrorDetails } from '../../../dto/xml-check.dto';
import { BaseRule } from '../base.rule';

export interface IXml3Rule {
  readonly key: string;
  readonly isEnabled: boolean;
  check(model: Xml3Model, context: HoSoContext): ErrorDetails | null;
}

export abstract class Xml3RuleBase extends BaseRule<Xml3Model> implements IXml3Rule {}
